### ISSUE TITLE
nn.GroupRevRes delivers 'DGLHeteroGraph' object has no attribute 'requires_grad'

### ISSUE THREAD

**Issue by jumxglhf** [dmlc/dgl#4176]
## To Reproduce

simply run the following code:

import dgl
import torch
import torch.nn as nn
from dgl.nn import GraphConv, GroupRevRes
class GNNLayer(nn.Module):
    def __init__(self, feats, dropout=0.2):
        super(GNNLayer, self).__init__()
        # Use BatchNorm and dropout to prevent gradient vanishing
        # In particular if you use a large number of GNN layers
        self.norm = nn.BatchNorm1d(feats)
        self.conv = GraphConv(feats, feats)
        self.dropout = nn.Dropout(dropout)
    def forward(self, g, x):
        x = self.norm(x)
        x = self.dropout(x)
        return self.conv(g, x)
num_nodes = 5
num_edges = 20
feats = 32
groups = 2
g = dgl.rand_graph(num_nodes, num_edges)
x = torch.randn(num_nodes, feats)
conv = GNNLayer(feats // groups)
model = GroupRevRes(conv, groups)
out = model(g, x)
out.sum().backward()

this code is identical to the provided examples except for the last line where I added a back propagation operation and the code crushes. 

## Environment

 - DGL Version: 0.8.2
 - Backend Library & Version PyTorch 1.11.0:

**Comment by mufeili** [comment]
Thank you for the report. I confirm the bug and will work on it.

**Comment by jumxglhf** [comment]
> This has been fixed in #4181 . Could you try building DGL from source with the source branch of the PR?

thanks for the timely update!
